shared/lib/error-utils.js

```diff
@@ -96,3 +96,61 @@ export async function getErrorHtml(errorKey, supportLink, metamaskState) {
     </div>
     `;
 }
+
+export async function getStateCorruptionErrorHtml(supportLink, metamaskState) {
+  let response, preferredLocale;
+  if (metamaskState?.currentLocale) {
+    preferredLocale = metamaskState.currentLocale;
+    response = await setupErrorLocale(metamaskState.currentLocale);
+  } else {
+    preferredLocale = await getFirstPreferredLangCode();
+    response = await setupErrorLocale(preferredLocale);
+  }
+
+  const textDirection = ['ar', 'dv', 'fa', 'he', 'ku'].includes(preferredLocale)
+    ? 'rtl'
+    : 'auto';
+
+  switchDirection(textDirection);
+  const { currentLocaleMessages, enLocaleMessages } = response;
+  const t = getLocaleContext(currentLocaleMessages, enLocaleMessages);
+  const hasBackup =
+    metamaskState?.data?.PreferencesController?.initializationFlags
+      ?.vaultBackedUp;
+  /**
+   * The pattern ${errorKey === 'troubleStarting' ? t('troubleStarting') : ''}
+   * is neccessary because we we need linter to see the string
+   * of the locale keys. If we use the variable directly, the linter will not
+   * see the string and will not be able to check if the locale key exists.
+   */
+  return `
+    <div class="critical-error__container">
+      <div class="critical-error">
+        <div class="critical-error__icon">
+          <svg width="24" height="24" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
+            <path d="m443 342l-126-241c-16-32-40-50-65-50-26 0-50 18-66 50l-126 241c-16 30-18 60-5 83 13 23 38 36 71 36l251 0c33 0 58-13 71-36 13-23 11-53-5-83z m-206-145c0-8 6-15 15-15 8 0 14 7 14 15l0 105c0 8-6 15-14 15-9 0-15-7-15-15z m28 182c-1 1-2 2-3 3-1 0-2 1-3 1-1 1-2 1-4 2-1 0-2 0-3 0-2 0-3 0-4 0-2-1-3-1-4-2-1 0-2-1-3-1-1-1-2-2-3-3-4-4-6-9-6-15 0-5 2-11 6-15 1 0 2-1 3-2 1-1 2-2 3-2 1-1 2-1 4-1 2-1 5-1 7 0 2 0 3 0 4 1 1 0 2 1 3 2 1 1 2 2 3 2 4 4 6 10 6 15 0 6-2 11-6 15z"/>
+          </svg>
+        </div>
+        <div>
+          <p>
+            ${
+              hasBackup === true
+                ? t('stateCorruptionDetectedWithBackup')
+                : t('stateCorruptionDetectedNoBackup')
+            }
+          </p>
+          <p class="critical-error__footer">
+            <span>${t('unexpectedBehavior')}</span>
+            <a
```

Comment:
There is an update for support link, we can have a separate ticket to integrate with the support modal
https://github.com/MetaMask/metamask-extension/pull/30415

wdyt？